performance_test.py

```diff
@@ -0,0 +1,175 @@
+import subprocess
+import datetime
+import csv
+import time
+import os
+import shutil
+import resource
+from dataclasses import dataclass
+import psutil
+import errno
+
+@dataclass
+class Result:
+    returncode: int
+    stdout: str
+    stderr: str
+    duration: float
+    rusage: resource.struct_rusage
+
+LOG_FILE = "probe_log"
+RECORD_DIR = "probe_record"
+
+class ResourcePopen(subprocess.Popen):
+    def _try_wait(self, wait_flags):
+        try:
+            (pid, sts, res) = os.wait4(self.pid, wait_flags)
+        except OSError as e:
+            if e.errno != errno.ECHILD:
+                raise
+            pid = self.pid
+            sts = 0
+        else:
+            self.rusage = res
+        return (pid, sts)
+
+def resource_call(*popenargs, timeout=None, **kwargs):
+    with ResourcePopen(*popenargs, **kwargs) as p:
+        try:
+            stdout, stderr = p.communicate(timeout=timeout)
+            return p.returncode, p.rusage, stdout, stderr
+        except:
+            p.kill()
+            stdout, stderr = p.communicate()
+            raise
```

Comment:
See my comment from the other PR about this function

Reply:
Additionally, can you print `result.stdout` and `result.stderr` and confirm the commands are running correctly?